File: 0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n=nums.length;
        int [] b=new int[n];
        b[0]=1;
        for(int i=1;i<n;i++){
           b[i]=b[i-1]*nums[i-1];
        }
        int s=1;
        for(int j=n-1;j>=0;j--){
            b[j]*=s;
            s*=nums[j];
        }
        return b;
    }
}

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution(object):
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        n=len(nums)
        a= [0]*n
        a[0]=1
        for i in range (1,n):
            a[i]=a[i-1]*nums[i-1]
        s=1
        for i in range( n-1,-1,-1):
            a[i]*=s
            s*=nums[i]
        return a

            
